Here are scripts for assemby and related analysis of T2T-sheep1.0.

Description for each directory:
The 01_Assembly directory contains codes for trio binning, haploid genome assembly. The commands for Hi-C anchoring and assemblies of telomeres are also shwon in the directory.

The 02_Polish directory contains codes for polish of T2T-sheep1.0.

The 03_Assessment directory contains codes for colinearity analysis compared with current reference, BUSCO evaluation, Quality Value (QV) assessment and coverage calculating of raw data (HiFi, ONT, NGS).

The 04_SD directory contains codes for SD calling and circos plot.

The 05_Methylation directory contains codes for methylation calling based on HiFi and ONT sequencing data. 

The 06_ChIP-seq directory contains codes for mapping and processing ChIP-seq data.

The 07_Annotation directory contains codes for repeat annotations, transcript-based prediction, de novo prediction and homolog-based prediction of genes.

The 08_Centromeres directory contains codes for sequence similarity calculating and visualization using StainedGlass, calculation of the sequence complexity and entropy using NeSSie and identification of satellite repeats using SRF.

The 09_OrthoFinder directory contains codes for identifying orthologous groups using OrthoFinder.

